modules/azure/src/main/clojure/xtdb/azure.clj

```diff
@@ -21,34 +22,43 @@
            [xtdb.api.log AzureEventHub AzureEventHub$Factory]
            [xtdb.api.storage AzureBlobStorage AzureBlobStorage$Factory]))
 
-(defmethod bp/->object-store-factory ::object-store [_ {:keys [storage-account container servicebus-namespace servicebus-topic-name prefix user-managed-identity-client-id]}]
+(defmethod bp/->object-store-factory ::object-store [_ {:keys [storage-account container servicebus-namespace servicebus-topic-name 
+                                                               prefix user-managed-identity-client-id
+                                                               storage-account-endpoint servicebus-namespace-fqdn]}]
   (cond-> (AzureBlobStorage/azureBlobStorage storage-account container servicebus-namespace servicebus-topic-name)
     prefix (.prefix (util/->path prefix))
-    user-managed-identity-client-id (.userManagedIdentityClientId user-managed-identity-client-id)))
+    user-managed-identity-client-id (.userManagedIdentityClientId user-managed-identity-client-id)
+    storage-account-endpoint (.storageAccountEndpoint storage-account-endpoint)
+    servicebus-namespace-fqdn (.serviceBusNamespaceFQDN servicebus-namespace-fqdn)))
 
 ;; No minimum block size in azure
 (def minimum-part-size 0)
 
-(defn open-object-store [^AzureBlobStorage$Factory factory]
-  (let [user-managed-identity-id (.getUserManagedIdentityClientId factory)
+(defn open-object-store [^AzureBlobStorage$Factory factory] 
+  (let [storage-account-endpoint (cond
```

Comment:
would be good to refactor this into the Kotlin so that we verify config ASAP (i.e. less time with potentially incorrect config), but this one can wait til post-merge.

Reply:
Yeah - there's a fair few things I'd LIKE to do on top of the kotlin config, but I'm not sure how easy the `kaml` serde stuff makes our life here..